R/bainCommon.R

```diff
@@ -0,0 +1,1215 @@
+    table$addColumnInfo(name = "PMP1",       type = "number", title = gettext("PMP a"))
+    table$addColumnInfo(name = "PMP2",       type = "number", title = gettext("PMP b"))
+    message <- gettext("BF.c denotes the Bayes factor of the hypothesis in the row versus its complement.\
+	Posterior model probabilities (a: excluding the unconstrained hypothesis, b: including the unconstrained hypothesis) are based on equal prior model probabilities.")
+  }
+  table$addFootnote(message = message)
+  table$addCitation(.bainGetCitations())
+  bainContainer[["mainResultsTable"]] <- table
+  
+  if (!ready || (type == "sem" && options[["model"]] == ""))
+    return()
+  
+  if(type %in% c("onesampleTTest", "independentTTest")){
+    
+    testType <- base::switch(options[["hypothesis"]],
+                             "equalNotEqual"       = 1,
+                             "equalBigger"         = 2,
+                             "equalSmaller"        = 3,
+                             "biggerSmaller"       = 4,
+                             "equalBiggerSmaller"  = 5)
+    
+    table$setExpectedSize(length(options[["variables"]]))
+    startProgressbar(length(options[["variables"]]))
+    for (variable in options[["variables"]]) {
+      
+      bainAnalysis <- .bainAnalysisState(dataset, options, bainContainer, ready, type, variable = variable, testType = testType)
+      
+      if (isTryError(bainAnalysis)){
+        table$addRows(list(Variable=variable), rowNames=variable)
+        table$addFootnote(message = gettextf("Results not computed: %1$s.", jaspBase:::.extractErrorMessage(bainAnalysis)), colNames = "Variable", rowNames = variable)
+        progressbarTick()
+        next
+      }
+      
+      if (variable %in% missing) {
+        table$addFootnote(message = gettext("Variable contains missing values, the rows containing these values are removed in the analysis."), colNames = "Variable", rowNames = variable)
+      }
+      
+      if (testType == 1) {
+        BF_0u <- bainAnalysis$fit$BF[1]
+        PMP_u <- bainAnalysis$fit$PMPb[2]
+        PMP_0 <- bainAnalysis$fit$PMPb[1]
+        if (options$bayesFactorType == "BF10")
+          BF_0u <- 1/BF_0u
+      }
+      if (testType == 2) {
+        BF_01 <- bainAnalysis$BFmatrix[1,2]
+        PMP_1 <- bainAnalysis$fit$PMPa[2]
+        PMP_0 <- bainAnalysis$fit$PMPa[1]
+        if (options$bayesFactorType == "BF10")
+          BF_01 <- 1/BF_01
+      }
+      if (testType == 3) {
+        BF_01 <- bainAnalysis$BFmatrix[1,2]
+        PMP_0 <- bainAnalysis$fit$PMPa[1]
+        PMP_1 <- bainAnalysis$fit$PMPa[2]
+        if (options$bayesFactorType == "BF10")
+          BF_01 <- 1/BF_01
+      }
+      if (testType == 4) {
+        BF_01 <- bainAnalysis$BFmatrix[1,2]
+        PMP_0 <- bainAnalysis$fit$PMPa[1]
+        PMP_1 <- bainAnalysis$fit$PMPa[2]
+        if (options$bayesFactorType == "BF10")
+          BF_01 <- 1/BF_01
```

Comment:
These seem identical... Might also be good to move to the calculation function, so when the results ever change we're aware that we have such a specific dependence on the results structure.

Reply:
Both done